controller/contour/controller.go

```diff
@@ -119,71 +121,88 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // ensureContour ensures all necessary resources exist for the given contour.
 func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour) error {
+	var errs []error
 	// Before doing anything with the contour, ensure it has a finalizer
 	// so it can cleaned-up later.
 	if err := r.ensureFinalizer(ctx, contour); err != nil {
-		return fmt.Errorf("failed to finalize contour %s/%s: %w", contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to finalize contour %s/%s: %w", contour.Namespace, contour.Name, err))
 	}
 	if err := r.ensureNamespace(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
-			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
+			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err))
 	}
 	if err := r.ensureRBAC(ctx, contour); err != nil {
```

Comment:
Is it useful to call `ensureRBAC` if there was an error while ensuring the namespace?  Some of the resources that `ensureRBAC` creates are namespaced, and the deployment and daemonset are namespaced, so calling `ensureRBAC` if the namespace does not exist generates additional errors and cannot produce a running Contour or Envoy instance.

Reply:
Understood. I took this approach so `syncContourStatus()` is called without requiring conditionals for the ensure namespace and rbac funcs. Any suggestions?

Reply:
It would make sense to me to use conditionals.  That is, `ensureNamespace` could return a Boolean value indicating whether or not the namespace exists, and the call to `ensureRBAC` could be wrapped in an `if haveNamespace {` block.  (This is how I approach "ensure" methods in general:  If resource Y requires resource X, then `ensureX` should return a Boolean return value `haveX`, and the call to `ensureY` should be wrapped in `if haveX {`.)